CDCgov/prime-reportstream
# 1911 - Fixed Patient_age display negative

This PR is to fix the patient's age bug which give negative value.  In addition to fix the bug, it also added the unit test 
for patient's age validation to make sure it is a good patient age to persist.

Test Steps:
1. Shutdown docker-compose
`docker-compose` down
2. Remove the prime-router_prime_dev container from the docker
`docker` container rm <prime-router_primer_dev container ID>
3.) Clone the repository and build prime-router_prime_dev docker image 
./gradlew clean package
3.) Start the docker-compose
docker-compose up.
4.) Run the following command to persist the report to the covid_result_metadata database table.
curl -X POST -H "client:simple_report" -H "content-type:text/csv" --data-binary "@./junk/PatientAgeTest.csv" "http://localhost:7071/api/reports"
4.) Note the report_id ("id") from the response resource.
5.) In Postgres Data users can query the table using the following:
SELECT patient_age from covid_result_metadata where report_id= '<noted id in step 4 above>'
user will NOT see the patient_age: Negative Number.

Test file: 
[PatientAgeTest.csv](https://github.com/CDCgov/prime-reportstream/files/7497898/PatientAgeTest.csv)

Screenshot:
![image](https://user-images.githubusercontent.com/20068283/140585128-e3093dd3-73c6-4d7a-ab1e-7732ab96c95f.png)


## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing
   ./gradlew test

### Process


## Fixes
- #issue - List GitHub issues this PR fixes
- Fixed patient_age display negative number ticket: https://app.zenhub.com/workspaces/prime-reportstream-5ff4833beb3e08001a4cacae/issues/cdcgov/prime-reportstream/1911

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
